### PR TITLE
Add workflow to check git conflict for backport PR

### PR DESCRIPTION
**Pull request by hlts2**
### Description

In backport PR, conflicts are forced to be included.
For this reason, I have implemented a workflow that checks conflicts in the diffs.

### Versions

- Vald Version: v1.7.12
- Go Version: v1.22.5
- Rust Version: v1.77.2
- Docker Version: v27.1.1
- Kubernetes Version: v1.30.3
- Helm Version: v3.15.3
- NGT Version: v2.2.4
- Faiss Version: v1.8.0

### Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/main/CONTRIBUTING.md) document and completed [our CLA agreement](https://cla-assistant.io/vdaas/vald).
- [x] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?

## Summary by CodeRabbit

- **New Features**
  - Introduced an automated workflow to check for merge conflicts in pull requests, enhancing collaboration and code integration.

- **Improvements**
  - Streamlined the pull request process by providing early detection of potential merge conflicts to prompt users for resolution.
  
- **Documentation**
  - Updated README to include a conflict marker, indicating an unresolved merge conflict that requires attention.

### CODE REVIEW

File: .github/workflows/check-conflict.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          token: ${{ secrets.DISPATCH_TOKEN }}
+      - name: Set Git config
+        run: |
+          git config --global --add safe.directory ${GITHUB_WORKSPACE}
+      - name: Check conflict
+        run: |
+          if grep -r ">>> HEAD" . --exclude-dir=.git --exclude=check-conflict.yml; then
```
Comment (kpango) [inline .github/workflows/check-conflict.yml:38]:
```suggestion
          if grep -r "<<<< HEAD" . --exclude-dir=.git --exclude=check-conflict.yml; then
```

File: .github/workflows/check-conflict.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        with:
+          fetch-depth: 0
+          token: ${{ secrets.DISPATCH_TOKEN }}
+      - name: Set Git config
+        run: |
+          git config --global --add safe.directory ${GITHUB_WORKSPACE}
+      - name: Check conflict
+        run: |
+          if grep -r "<<<< HEAD" . --exclude-dir=.git --exclude=check-conflict.yml; then
+            echo "Please fix conflict locally."
```
Comment (kpango) [inline .github/workflows/check-conflict.yml:39]: how about post message to PR?

Comment (hlts2) [reply .github/workflows/check-conflict.yml:51]: It looks good!! Thank you!
I have added that feature!

**Review by kmrmt — APPROVED** [review]

**Review by kpango — APPROVED** [review]
LGTM thanks!!

**Pull request merged**